subset/network/network_tests.py

```diff
@@ -0,0 +1,91 @@
+import subprocess
+import time
+import sys
+
+name_of_tests = ['network.min_send', 'network.application.min_send', 'dhcp.long', 'ntp.update']
+
+arguments = sys.argv
+
+test_request = str(arguments[1])
+
+test_id = -1
+
+for x in range(0, len(name_of_tests) - 1):
+    if test_request == name_of_tests[x]:
+        test_id = x
+
+print('test_id=' + str(test_id))
+
+cap_pcap_file = str(arguments[2])
+#capture.pcap
+
+device_address = '127.0.0.1'
+
+if test_id == 0:
+    device_address = str(arguments[3])
+
+report_filename = 'report.txt'
+
+packets_in_report = 10
+
+tcpdump_display_all_packets = 'tcpdump -n src host ' + device_address + ' -r ' + cap_pcap_file
+tcpdump_display_udp_bacnet_packets = 'tcpdump -n udp dst portrange 47808-47809 ' + cap_pcap_file
+tcpdump_display_arp_packets = 'tcpdump -v arp -r ' + cap_pcap_file
+tcpdump_display_ntp_packets = 'tcpdump dst port 123 -r ' + cap_pcap_file
+
+tests = [tcpdump_display_all_packets, tcpdump_display_udp_bacnet_packets, tcpdump_display_arp_packets, tcpdump_display_ntp_packets]
+
+def shell_command_fb(command, wait_time, terminate_flag):
+    process = subprocess.Popen(command, universal_newlines=True, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    text = process.stdout.read()
+    retcode = process.wait()
+    time.sleep(wait_time)
+    if terminate_flag:
+        process.terminate()
+    if len(text) > 0:
+        return text
+
+pointer_list_line_end = []
+
+def find_char_pointer(value, find_text, pointer_list):
+    pointer = 0
```

Comment:
this is an index, not a pointer

Reply:
Its a memory location in data.
https://en.wikipedia.org/wiki/Pointer_(computer_programming)

Reply:
Pointers are references to a memory location, this is an [index into an array](https://en.wikipedia.org/wiki/Pointer_(computer_programming)#C_arrays). Python and Java don't really expose pointers. Search that link you sent me for Java and you'll see some discussion about the difference -- and notice that Python isn't mentioned at all.

Reply:
Function "find_char_pointer" is finding characters pointers in a single string(byte array) and passing to another Function "cut_packets_to_list" to allow the string(byte array) to be split into a packet_list array.

Reply:
You are fundamentally confusing a "pointer" with an "index" is, and they are not the same thing. If you look at [the definition of the "find" function](https://www.geeksforgeeks.org/string-find-python/) which you are using: "The find() method returns the lowest index of the substring if it is found in given string. If its is not found then it returns -1."

Do you have any reason why "index" is not appropriate here?

Reply:
Im not confused at all. Im working with bytes in my head as I;m coding it. Just because python uses index. doesnt mean im confused.

If what your saying is that "index" is easier for a "python" scripter to understand then I finally see the point, and will change to reflect. But up until now I write software for control systems and control systems are written in c || c++ bits and bytes

Reply:
It's not python, it's true in C/C++ as well. Consider [strstr](http://www.cplusplus.com/reference/cstring/strstr/), which explicitly returns a `char *` _pointer_ to a string, which is different than something like [string::find](https://www.geeksforgeeks.org/string-find-in-cpp/), which returns [an integer index](https://en.wikipedia.org/wiki/Array_data_structure). It _not_ correct to say something `a[strstr(a, "b")]`, since `strstr` returns a _pointer_, while `a[n]` requires an _index_. Functionally, if ti's an index then you need the base object (`a[i]`) to access the data, where with a _pointer_ you do not (it's just `*p`). Thus the common use of the variable `i` for loops, and `p` for pointers.

I'd be curious if you had documentation for something (in whatever language) that is calling the relative position of an object in an array a_pointer_ rather than _index_ or _position_.